pkg/controller/vitessshard/reconcile_backup_job.go

```diff
@@ -219,31 +220,35 @@ func vtbackupSpec(key client.ObjectKey, vts *planetscalev2.VitessShard, parentLa
 	minRetentionTime := time.Duration(0)
 	minRetentionCount := 1
 
+	// Fill in the parts of a vttablet spec that make sense for vtbackup.
+	tabletSpec := &vttablet.Spec{
+		GlobalLockserver:         vts.Spec.GlobalLockserver,
+		Labels:                   labels,
+		Images:                   vts.Spec.Images,
+		KeyRange:                 vts.Spec.KeyRange,
+		Vttablet:                 &pool.Vttablet,
+		Mysqld:                   pool.Mysqld,
+		DataVolumePVCName:        key.Name,
+		DataVolumePVCSpec:        pool.DataVolumeClaimTemplate,
+		KeyspaceName:             keyspaceName,
+		DatabaseName:             vts.Spec.DatabaseName,
+		DatabaseInitScriptSecret: vts.Spec.DatabaseInitScriptSecret,
+		BackupLocation:           backupLocation,
+		BackupEngine:             vts.Spec.BackupEngine,
+		InitContainers:           pool.InitContainers,
+		SidecarContainers:        pool.SidecarContainers,
+		ExtraEnv:                 pool.ExtraEnv,
+		Annotations:              vts.Spec.Annotations,
```

Comment:
The field `vts.Spec.Annotations` is for annotations on the VitessShard object. I think what we want is `pool.Annotations`?

Also, did you check that this field is respected for vtbackup Pods even though it wasn't being filled in before?

Reply:
Done. Also, I'm not sure what you mean by `respected`. Would you mind re-phrasing that question?

Reply:
The code that builds vtbackup Pods is separate from the code that builds vttablet Pods. Only some of these vttablet spec fields are relevant to vtbackup Pods, so by default a field added here will have no effect on vtbackup Pods unless we go fix the vtbackup Pod code to look at the new field.